src/features/announcements/Announcement.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useContext } from "react";
+import { Button } from "components/ui/Button";
+import { Context } from "features/game/GameProvider";
+import { announcements } from "features/announcements";
+import fence from "assets/land/goblin_fence.png";
+import { useShowScrollbar } from "lib/utils/hooks/useShowScrollbar";
+import classNames from "classnames";
+
+const CONTENT_HEIGHT = 400;
+export const Announcement: React.FC = () => {
+  const { ref: itemContainerRef, showScrollbar } =
+    useShowScrollbar(CONTENT_HEIGHT);
+  const { gameService } = useContext(Context);
+
+  const storedDate = localStorage.getItem("announcementLastRead");
+
+  function onAcknowledge() {
+    gameService.send("ACKNOWLEDGE");
+  }
+
+  const filtered = storedDate
+    ? announcements.filter(
+        (announcement) => announcement.date > new Date(storedDate)
+      )
+    : announcements;
+
+  // Sort latest first
+  filtered.sort((a, b) => b.date.getTime() - a.date.getTime());
```

Comment:
Business logic looks good. I would suggest abstracting the storage logic into a separate file. This helps keep the React component focussed solely on the UI.

e.g.

```
const announcements = getNewAnnouncements()
```

Where `getNewAnnouncements` encapsulates the localstorage and filtering functionality.